### PR TITLE
Implemented a verification procedure for ProjectActivityScore

### PR DESCRIPTION
**Pull request by artem-smotrakov**
Main updates:

- Updated `Rating.allFeatures()` to return a set of features.
- Updated and added several methods to use collections of features.
- Added `ProjectActivityScore.Verification` class.
- Added a `ProjectActivityScoreTestVectors` notebook which defines the test vectors.
- Added `ProjectActivityScoreVerification` test which runs the verification.
- Fixed checkstyle warnings in the modified files.

### CODE REVIEW

File: src/main/java/com/sap/sgs/phosphor/fosstars/model/rating/AbstractRating.java [hunk 1]
```diff
@@ -102,19 +113,18 @@ public boolean equals(Object o) {
       return false;
     }
     AbstractRating that = (AbstractRating) o;
-    return Objects.equals(score, that.score) &&
-        version == that.version;
+    return Objects.equals(score, that.score) && version == that.version;
```
Comment (Sachpat) [inline src/main/java/com/sap/sgs/phosphor/fosstars/model/rating/AbstractRating.java]: Just a question, why don't we use Objects.equals(version, that.version) here?

Comment (artem-smotrakov) [reply src/main/java/com/sap/sgs/phosphor/fosstars/model/rating/AbstractRating.java]: `version` is a enum, the simplest way to compare them is `==`.

**Comment by artem-smotrakov** [comment]
Thanks for the review @Sachpat !

File: src/main/java/com/sap/sgs/phosphor/fosstars/model/rating/AbstractRating.java (hunk 1 above)
Comment (Sachpat) [reply src/main/java/com/sap/sgs/phosphor/fosstars/model/rating/AbstractRating.java]: Aah missed that it's an enum :D thanks for the clarification!

**Pull request merged**